tests/test_send_recv_two_workers.py

```diff
@@ -0,0 +1,185 @@
+            # cdf = cudf.DataFrame({"a": range(5), "b": range(5)}, index=[10,13,15,20,25])
+            size = 2**12
+            cdf = cudf.DataFrame({"a": np.random.random(size), "b": np.random.random(size)}, index=np.random.randint(size, size=size))
+            # cdf = cudf.DataFrame({"a": range(5), "b": [1.2, None, 2, 3, 4]}, index=[1, 2, 5, None, 6])
+            print(cdf.head().to_pandas())
+            header, _frames = cdf.serialize()
+            frames = [pickle.dumps(header)] + _frames
+
+    return frames
+
+
+def server(env, port):
+    # create listener receiver
+    # write cudf object
+    # confirm message is sent correctly
+
+    os.environ.update(env)
+    print("ENV SERVER: ", os.environ)
+
+    async def f(lisitener_port):
+        ucp.init()
+
+        # coroutine shows up when the client asks
+        # to connect
+        async def write(ep, li):
+            await asyncio.sleep(0.1)
+            frames = await generate_cuda_obj("cudf")
+            print("Generated CUDA Data")
+            # print(frames)
+            is_gpus = b"".join(
+                [
+                    struct.pack("?", hasattr(frame, "__cuda_array_interface__"))
+                    for frame in frames
+                ]
+            )
+            nframes = struct.pack("Q", len(frames))
+            sizes = b"".join([struct.pack("Q", nbytes(frame)) for frame in frames])
+            meta = b"".join([nframes, is_gpus, sizes])
+
+            print("Sending meta...")
+            await ep.send_obj(meta)
+            for idx, frame in enumerate(frames):
+                await ep.send_obj(frame)
+
+            print("CONFIRM RECEIPT")
+            resp = await ep.recv_future()
+            obj = ucp.get_obj_from_msg(resp)
+            assert obj == b"OK"
+            print("Shutting Down Server...")
+            ucp.stop_listener(li)
+            ucp.fin()
+
+        loop = asyncio.get_event_loop()
+        listener = ucp.start_listener(
+            write, listener_port=lisitener_port, is_coroutine=True
+        )
+        t = loop.create_task(listener.coroutine)
+        await t
+        print("Last Bit of Cleanup...")
+        ucp.fin()
+
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(f(port))
+
+
+def test_send_recv_cudf(event_loop):
+    base_env = {
+        "UCX_RNDV_SCHEME": "put_zcopy",
+        "UCX_MEMTYPE_CACHE": "n",
+        "UCX_TLS": "rc,cuda_copy,cuda_ipc",
+        "CUDA_VISIBLE_DEVICES": "6,2",
+    }
+    env1 = base_env.copy()
+    env2 = base_env.copy()
+    env2["CUDA_VISIBLE_DEVICES"] = "2,6"
+
+    port = 15338
+    server_process = multiprocessing.Process(
+        name="server", target=server, args=[env1, port]
+    )
+    client_process = multiprocessing.Process(
+        name="client", target=client, args=[env2, port]
+    )
+
+    server_process.start()
+    client_process.start()
+
+    server_process.join()
+    client_process.join()
+
+    ucp.fin()
```

Comment:
`ucp.fin()` is already [here](https://github.com/rapidsai/ucx-py/pull/174/files#diff-1cca3066e91b8737cf0377fe360bbfd3R145) and [here](https://github.com/rapidsai/ucx-py/pull/174/files#diff-1cca3066e91b8737cf0377fe360bbfd3R70). Do we need this? I guess calling `fin()` after `fin()` isn't erroneous but maybe not call that because we don't reference ucp in this function at all?